github-status/handler_test.go

```diff
@@ -8,75 +8,89 @@ import (
 	"github.com/openfaas/openfaas-cloud/sdk"
 )
 
-func TestBuildURLWithoutPrettyURL_WithSlash(t *testing.T) {
-	os.Setenv("gateway_public_url", "http://localhost:8080")
-	os.Setenv("gateway_pretty_url", "")
+func Test_buildPublicStatusURL(t *testing.T) {
 
-	event := &sdk.Event{
-		Owner:   "alexellis",
-		Service: "tester",
-	}
-
-	val := buildPublicStatusURL("success", sdk.BuildFunctionContext(event.Service), event)
-	want := "http://localhost:8080/function/alexellis-tester"
-
-	if val != want {
-		t.Errorf("building PublicURL: want %s, got %s", want, val)
-		t.Fail()
-	}
-}
-
-func TestBuildURLWithoutPrettyURL_WithOutSlash(t *testing.T) {
-	os.Setenv("gateway_public_url", "http://localhost:8080")
-	os.Setenv("gateway_pretty_url", "")
-
-	event := &sdk.Event{
-		Owner:   "alexellis",
-		Service: "tester",
-	}
-
-	val := buildPublicStatusURL("success", sdk.BuildFunctionContext(event.Service), event)
-	want := "http://localhost:8080/function/alexellis-tester"
-
-	if val != want {
-		t.Errorf("building PublicURL: want %s, got %s", want, val)
-		t.Fail()
-	}
-}
-
-func TestBuildURLWithPrettyURL(t *testing.T) {
-	os.Setenv("gateway_public_url", "http://localhost:8080")
-	os.Setenv("gateway_pretty_url", "https://user.openfaas-cloud.com/function")
-
-	event := &sdk.Event{
-		Owner:   "alexellis",
-		Service: "tester",
-	}
-
-	val := buildPublicStatusURL("success", sdk.BuildFunctionContext(event.Service), event)
-	want := "https://alexellis.openfaas-cloud.com/tester"
-
-	if val != want {
-		t.Errorf("building PublicURL: want %s, got %s", want, val)
-		t.Fail()
+	tests := []struct {
+		title            string
+		status           string
+		gatewayPublicUrl string
+		gatewayPrettyUrl string
+		eventOwner       string
+		eventService     string
+		eventURL         string
+		want             string
+	}{
+		{
+			title:            "Undefined status gives original URL",
+			status:           "not-supported",
+			gatewayPublicUrl: "",
+			gatewayPrettyUrl: "",
+			eventOwner:       "alexellis",
+			eventService:     "tester",
+			eventURL:         "http://original-value.local",
+			want:             "http://original-value.local",
+		},
+		{
+			title:            "With pretty URL",
+			status:           "success",
+			gatewayPublicUrl: "http://localhost:8080",
```

Comment:
I think the dashboard cannot be accessed with localhost, I mean github or gitlab cannot redirect it correct me if I am wrong